HARK/ConsumptionSaving/ConsPortfolioModel.py

```diff
@@ -429,6 +431,39 @@ def solve_one_period_ConsPortfolio(
     RiskyMax = np.max(Risky_next)
     RiskyMin = np.min(Risky_next)
 
+    # Perform an alternate calculation of the absolute patience factor when
+    # returns are risky. This uses the Merton-Samuelson limiting risky share,
+    # which is what's relevant as mNrm goes to infinity.
+    def calc_Radj(R):
+        Rport = ShareLimit * R + (1.0 - ShareLimit) * Rfree
+        return Rport ** (1.0 - CRRA)
+
+    R_adj = expected(calc_Radj, RiskyDstn)[0]
+    PatFac = (DiscFacEff * R_adj) ** (1.0 / CRRA)
+    MPCminNow = 1.0 / (1.0 + PatFac / solution_next.MPCmin)
+
+    # Also perform an alternate calculation for human wealth under risky returns
+    def calc_hNrm(S):
+        Risky = S["Risky"]
+        PermShk = S["PermShk"]
+        TranShk = S["TranShk"]
+        G = PermGroFac * PermShk
+        Rport = ShareLimit * Risky + (1.0 - ShareLimit) * Rfree
+        hNrm = (G / Rport**CRRA) * (TranShk + solution_next.hNrm)
+        return hNrm
+
+    # This correctly accounts for risky returns and risk aversion
+    hNrmNow = expected(calc_hNrm, ShockDstn) / R_adj
```

Comment:
It's very specifically *not* that, and that's the point of this PR. To get the limiting linear cFunc correct, the code I included for `hNrm` is what needs to be done. It *very specifically* computes `R_adj` outside of that statement and applies it outside it as well.

When `RiskyShare` is chosen optimally, the first order condition of the optimization problem in `calc_limiting_share` means that we *could* just discount by `Rfree` rather than use the more complicated calculation.

Reply:
Ok, I don't know how to derive this equation, but it just made intuitive sense to me that if both the nominator and denominator (`h_nrm`, `R_adj`) include the rv. `risky`, then we should not be evaluating the expectations separately. 

From reading your emails, it seemed you don't quite have a proof for this? Is it possible to at least get a visual proof? How did you figure out it's this expression and not something else?

Reply:
That's correct, I don't have a proof. I came up with it by a combination of trial and error and intuition from what I was seeing as I tried different parameter values:

1) The "hNrm error" was directly related to return risk: bigger `RiskyStd` made the difference between calculated and "correct" hNrm bigger.

2) The sign of the error depended on whether `CRRA` was greater or lesser than 1. When `CRRA=1`, the extent of return risk had no effect and there was no error. The further from 1 that rho was, the larger the error (fixing riskiness of returns).

So I knew there had to be something that canceled out for all values of rho when there's no return risk, and made return risk not matter when rho=1. I then just experimented with raising Rport to different powers that sum (or difference) to 1 and have power=0 when rho=1 for one of the powers. I eventually got lucky.

It was a good thing I did that work with the `RiskyAsset` model, else I would have stumbled onto the `Rfree` thing in the portfolio model and then been stumped on the other ones.

I can make "visual proof by examples", which aren't proofs at all. They involve solving the problem out to extremely high values of aNrm like 1B and show that the numerically constructed cFunc *really does* converge to the limiting linear function defined by MPCmin and hNrm.